src/Symfony/Component/Routing/Loader/AttributeClassLoader.php

```diff
@@ -227,6 +227,11 @@ protected function addRoute(RouteCollection $collection, object $attr, array $gl
                     }
                     break;
                 }
+
+                if (preg_match(\sprintf('/\{(?<argName>[^\:]+):%s(?:(\.[^\}]+)?)\}/', preg_quote($param->name)), $path, $matches)) {
```

Comment:
```suggestion
                if (preg_match(\sprintf('/\{(?<argName>[^\:]++):%s(?:(\.[^\}]++)?)\}/', preg_quote($param->name)), $path, $matches)) {
```

Reply:
also `(?:<.*?>)?` before the closing `}` like above?

Reply:
Are the mapped parameters avaiable with Enum Values or just entities ?

Reply:
Since I thought it was only for entities, I excluded Enums

Reply:
I think we can merge both calls to preg_match in one:
```suggestion
                if (preg_match(\sprintf('/\{(?|([^\:]++):%s(?:\.[^\}<]++)?|(%1$s))(?:<.*?>)?\}/', preg_quote($param->name)), $path, $matches)) {
```

`$matches[1]` should contain the correct name of the routing param